src/core/gui/toolbarMenubar/ToolPageLayer.cpp

```diff
@@ -123,26 +117,21 @@ void ToolPageLayer::layerMenuShowClicked(GtkWidget* menu) {
         return;
     }
 
-    int layerId = -1;
-
-    for (auto& kv: showLayerItems) {
-        if (kv.second == menu) {
-            layerId = kv.first;
-            break;
-        }
-    }
+    auto it = std::find_if(showLayerItems.begin(), showLayerItems.end(),
+                           [menu](std::pair<const size_t, GtkWidget*>& a) { return a.second == menu; });
 
-    if (layerId < 0) {
+    if (it == showLayerItems.end()) {
         g_warning("Invalid Layer Show Menu selected - not handled");
         return;
     }
+    auto layerId = it->first;
```

Comment:
Replace with std::find. Please double check what I wrote doesn't change the behaviour.

Reply:
You might want to create a lambda variable for the binary_operator like `auto menu_finder = [menu](std::pair<const size_t, GtkWidget*>& a) { return a.second == menu; };`

Since you require it more than once.